fractal_tasks_core/tasks/apply_registration_to_ROI_tables.py

```diff
@@ -0,0 +1,312 @@
+
+logger = logging.getLogger(__name__)
+
+
+@validate_arguments
+def apply_registration_to_ROI_tables(
+    *,
+    # Fractal arguments
+    input_paths: Sequence[str],
+    output_path: str,
+    component: str,
+    metadata: dict[str, Any],
+    # Task-specific arguments
+    roi_table: str = "FOV_ROI_table",
+    reference_cycle: int = 0,
+    new_roi_table: str = None,
+) -> dict[str, Any]:
+    """
+    Applies pre-calculated registration to ROI tables.
+
+    Applies pre-calculated registration such that resulting ROIs contain
+    the consensus align region between all cycles.
+
+    This task runs on the well level.
+
+    Args:
+        input_paths: List of input paths where the image data is stored as
+            OME-Zarrs. Should point to the parent folder containing one or many
+            OME-Zarr files, not the actual OME-Zarr file. Example:
+            `["/some/path/"]`. This task only supports a single input path.
+            (standard argument for Fractal tasks, managed by Fractal server).
+        output_path: This parameter is not used by this task.
+            (standard argument for Fractal tasks, managed by Fractal server).
+        component: Path to the OME-Zarr image in the OME-Zarr plate that is
+            processed. Example: `"some_plate.zarr/B/03/0"`.
+            (standard argument for Fractal tasks, managed by Fractal server).
+        metadata: Dictionary containing metadata about the OME-Zarr. This task
+            does not use the metadata.
+        roi_table: Name of the ROI table over which the task loops to
+            calculate the registration. Examples: `FOV_ROI_table` => loop over
+            the field of views, `well_ROI_table` => process the whole well as
+            one image.
+        reference_cycle: Which cycle to register against. Defaults to 0,
+            which is the first OME-Zarr image in the well, usually the first
+            cycle that was provided
+        new_roi_table: Optional name for the new, registered ROI table. If no
+            name is given, it will default to "registered_" + `roi_table`
+
+    """
+    if not new_roi_table:
+        new_roi_table = "registered_" + roi_table
+    logger.info(
+        f"Running for {input_paths=}, {component=}. \n"
+        f"Applying translation registration to {roi_table=} and store it as "
+        f"{new_roi_table=}."
+    )
+
+    well_zarr = f"{input_paths[0]}/{component}"
+    with open(f"{well_zarr}/.zattrs", "r") as jsonfile:
+        zattrs = json.load(jsonfile)
+
+    acquisition_dict = get_acquisition_paths(zattrs)
+    logger.info(
+        "Calculating common registration for the following cycles: "
+        f"{acquisition_dict}"
+    )
+
+    # TODO: Allow a filter on which acquisitions should get processed?
+
+    # Collect all the ROI tables
+    roi_tables = {}
+    for acq in acquisition_dict.keys():
+        curr_ROI_table = ad.read_zarr(f"{well_zarr}/{acq}/tables/{roi_table}")
```

Comment:
Something is unclear to me re:
```python
    for acq in acquisition_dict.keys():
        curr_ROI_table = ad.read_zarr(f"{well_zarr}/{acq}/tables/{roi_table}")
```

If I understand correctly, `acquistion_dict` has keys that are the integer acquisition indices, and values which are the corresponding paths. Shouldn't we use the values here, rather than the keys?

That is, I would expect something like
```python
    for acq in acquisition_dict.keys():
        acq_path = acquisition_dict[acq]
        curr_ROI_table = ad.read_zarr(f"{well_zarr}/{acq_path}/tables/{roi_table}")
```

Reply:
This is obviously irrelevant if `path=str(index)`, but it could become a problem if we set e.g. `path = f"cycle{index}"`.

Reply:
Good catch! Fixed now :)